code/modules/projectiles/guns/projectile/automatic.dm

```diff
@@ -26,6 +28,35 @@
 		list(mode_name="full auto",		can_autofire=1, burst=1, fire_delay=5, fire_delay_wielded=2, one_hand_fa_penalty=12, burst_accuracy = list(0,-1,-1,-2,-2,-2,-3,-3), dispersion = list(5, 10, 15, 20, 25))
 		)
 
+/obj/item/gun/projectile/automatic/update_icon()
+	..()
+	icon_state = (ammo_magazine)? "saber" : "saber-empty"
+
+/obj/item/gun/projectile/automatic/mounted
+	name = "mounted prototype SMG"
+	can_suppress = FALSE
+	auto_eject = TRUE
+	auto_eject_sound = 'sound/weapons/smg_empty_alarm.ogg'
+
+/obj/item/gun/projectile/automatic/mounted/load_ammo(obj/item/A, mob/user)
+	var/obj/item/rig/rig = get_rig()
+	if(istype(rig))
+		if(!rig.offline && rig.suit_is_deployed())
+			..()
+		else
+			to_chat(user, SPAN_WARNING("You can't reload your submachine gun without deploying your hardsuit!"))
+			balloon_alert(user, "Deploy your hardsuit first!")
+			return
+
+/obj/item/gun/projectile/automatic/mounted/unload_ammo(mob/user, allow_dump = 0)
+	var/obj/item/rig/rig = get_rig()
+	if(istype(rig))
+		if(!rig.offline && rig.suit_is_deployed())
+			..()
+		else
+			to_chat(user, SPAN_WARNING("You can't unload your submachine gun without deploying your hardsuit!"))
+			balloon_alert(user, "Deploy your hardsuit first!")
```

Comment:
this being copypasted on every mounted gun (along with the codeblock above) is not acceptable